### PR TITLE
Plugin core/plugin test

### PR DESCRIPTION
**Pull request by lks-mtzn**
### All changes meet the following requirements
- [x] Changelog entry was added
- [x] Changes have been tested
- [x] Plugin can be built

@plentymarkets/ceres-io

### CODE REVIEW

File: tests/Feature/CheckoutServiceCurrencyConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    /** @test */
+    public function check_method_get_currency_webstore_config()
+    {
+
+        /** @var PluginSet $pluginSet */
+        $pluginSet = factory(PluginSet::class)->create();
+
+        factory(Webstore::class)->create(
```
Comment (ggottwald) [inline tests/Feature/CheckoutServiceCurrencyConfigTest.php:59]: TODO:

File: tests/Feature/CheckoutServiceCurrencyConfigTest.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        $this->checkoutService = pluginApp(CheckoutService::class);
+
+    }
+
+    /** @test */
+    public function check_method_get_currency_session_storage()
+    {
+
+        $expectedCurrency = "USD";
```
Comment (ggottwald) [inline tests/Feature/CheckoutServiceCurrencyConfigTest.php]: TODO:

File: tests/TestCase.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        $contact = factory(Contact::class)->create(['blocked' => 0, 'password' => pluginApp('hash')->make($password)]);
+
+        factory(ContactOption::class)->create([
+            'contactId' => $contact->id,
+            'typeId' => ContactOption::TYPE_MAIL,
+            'subTypeId' => ContactOption::SUBTYPE_PRIVATE,
+            'value' => $email
+        ]);
+
+        /** @var AccountAuthenticationProxy $authProxy */
```
Comment (ggottwald) [inline tests/TestCase.php]: Move to test suite

File: tests/Unit/Repositories/ItemWishListGuesRepositoryTest.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
+<?php
```
Comment (ggottwald) [inline tests/Unit/Repositories/ItemWishListGuesRepositoryTest.php]: Gues*t*

File: tests/Unit/Repositories/ItemWishListGuesRepositoryTest.php [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            ->once()
+            ->andReturn(null);
+
+        $response = $this->wishListGuestRepository->removeItemWishListEntry(1000);
+
+        $this->assertTrue($response);
+    }
+
+    /** @test */
+    public function it_deletes_an_item_from_the_wish_list_wish_is_not_in_it()
```
Comment (ggottwald) [inline tests/Unit/Repositories/ItemWishListGuesRepositoryTest.php]: which

File: tests/Unit/Repositories/ItemWishListRepositoryTest.php [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+    protected $customerServiceMock;
+
+    /**
+     *
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->databaseMock = Mockery::mock(DataBase::class);
```
Comment (ggottwald) [inline tests/Unit/Repositories/ItemWishListRepositoryTest.php]: move to testsuite

File: tests/Unit/Services/CheckoutServiceCurrencyTest.php [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+        $this->checkoutService = pluginApp(CheckoutService::class);
+
+    }
+
+    /** @test */
+    public function it_returns_the_currency_from_session_storage()
+    {
+
+        $expectedCurrency = "USD";
```
Comment (ggottwald) [inline tests/Unit/Services/CheckoutServiceCurrencyTest.php]: Faker?

File: tests/Unit/Services/CustomerServiceTest.php [hunk 8]
```diff
@@ -0,0 +1,723 @@
...
+        app()->instance(ContactAccountRepositoryContract::class, $this->contactAccountRepositoryMock);
+
+        $this->sessionStorageServiceMock = Mockery::mock(SessionStorageService::class);
+        app()->instance(SessionStorageService::class, $this->sessionStorageServiceMock);
+
+        $this->webstoreConfigServiceMock = Mockery::mock(WebstoreConfigurationService::class);
+        app()->instance(WebstoreConfigurationService::class, $this->webstoreConfigServiceMock);
+
+        $this->dispatcherMock = Mockery::mock(Dispatcher::class);
+        app()->instance(Dispatcher::class, $this->dispatcherMock);
```
Comment (ggottwald) [inline tests/Unit/Services/CustomerServiceTest.php]: testsuite

**Review by Dominik2809 — DISMISSED** [review]

**Review by felixdausch — DISMISSED** [review]

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by lks-mtzn** [comment]
Restart tests.

**Comment by lks-mtzn** [comment]
Restart tests.

**Comment by lks-mtzn** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Comment by mdierkes** [comment]
Restart tests.

**Review by mdierkes — APPROVED** [review]

**Pull request merged**